fetch/fetch.go

```diff
@@ -612,6 +620,9 @@ func (f *Fetch) getHash(ctx context.Context, hash types.Hash32, h datastore.Hint
 
 // RegisterPeerHashes registers provided peer for a list of hashes.
 func (f *Fetch) RegisterPeerHashes(peer p2p.Peer, hashes []types.Hash32) {
+	if peer == f.host.ID() {
+		return
```

Comment:
this should be a programmer mistake. can you please log an .Error ?

Reply:
this is actually not a programmer mistake. when we publish an object (atx/proposal), the object go through the gossip handlers, which calls `RegisterPeerHashes()` on the peer.

alternatively i can add a check in those gossip handlers to not call `RegisterPeerHashes` if it's self-publish data. but i figure it's simpler to simply add the check here.

Reply:
my bad, i thought code is registering that stuff only for data that is not available locally